### DISCUSSION TITLE
I need to ignore the mouse for the window

### DISCUSSION THREAD

**Discussion by bedna-KU** [bbidulock/icewm#598] [Q&A]
I have one window (with terminal) maximized.
Second small window with 50% transparent above the first window.
I need to make small window ignore the mouse.

Attached screenshot from Awesome WM.
![osd](https://user-images.githubusercontent.com/4377084/120103464-f90eb580-c14f-11eb-98fa-960671d539dc.png)

**Comment by gijsbers** [comment]
You don't explain if these windows are related to the same application,
or the degree to which you can control them. I would look at the X11
extensions for composite, damage, fixes. Search for "X11 overlays".
Obviously you can't modify the event mask for some other application.

**Reply by bedna-KU** [reply]
The output is not from the same application.
I use compton for compositing.

**Reply by gijsbers** [reply]
It's not clear from your description if you want to hack the video app or not.
Maybe you can open an input-only window in override_redirect mode
on top of the video window and then forward some events to the other window
appropriately modified to account for the changes in geometry?

**Reply by bedna-KU** [reply]
I need ignore mouse on window with video.
When I click on the video window, I want the cursor to focus on the lower window with Vim,
Just make it as behaved the video isn't there.

**Reply by gijsbers** [reply]
And you cannot program the software?
Something could be added to the WM,
but it would be a fair amount of work.

**Reply by bedna-KU** [reply]
> And you cannot program the software?

I want it to work with any software 

> Something could be added to the WM, ...

I love IcewM, it would be good if it was there but I will probably survive without it.

**Reply by gijsbers** [reply]
Technically, you could let the video shine through the editor.
Make the editor opaque, but only where there is another
window behind it. That way all input would go to the editor.

**Reply by bedna-KU** [reply]
I know about this solution, but it's not it ...

**Reply by bedna-KU** [reply]
I have one idea, I'll try it tomorrow and let you know

**Reply by bedna-KU** [reply]
My solution.

To **.icewm/keys**
add line: `key "Ctrl+Alt+j"        $HOME/.icewm/contrib/jumping.sh`

Make script **.icewm/contrib/jumping.sh**
Make script executable.

Hover over the window you want to jump and press **Ctrl + J**. 
For end of script press again **Ctrl+J**

One picture is more than a hundred words 
![icewm_jumping_window](https://user-images.githubusercontent.com/4377084/120697397-f55f9380-c4ad-11eb-9425-d99d097dd92f.gif)

Script **jumping.sh**
```
#!/bin/bash

xaxis=$(xrandr --current | grep '*' | awk '{print $1}' |  cut -d 'x' -f1 | uniq)
font_width=`expr $xaxis / 40`

FILE="$HOME/.icewm/contrib/jumping_id.txt"

if test -f "$FILE"; then
    id=$(cat "$FILE")
    string="Disable jumping window $id"
    rm $FILE
    echo $string | osd_cat -A center -p middle -f -adobe-helvetica-bold-r-*-*-$font_width-120-*-*-*-*-*-* -cgreen -d1 -O3 &
else
    id=$(xdotool getmouselocation | awk -F'window:' '{print $2}')
    string="Start jumping window $id"
    echo $id >"$HOME/.icewm/contrib/jumping_id.txt"
    echo $string | osd_cat -A center -p middle -f -adobe-helvetica-bold-r-*-*-$font_width-120-*-*-*-*-*-* -cgreen -d1 -O3

while test -f "$FILE"; do
    id_saved=$(cat "$FILE")
    # echo "SAVED  "$id_saved
    id_hex=$(printf 0x%x $id_saved)
    # echo $id_hex
    # echo "XXX "$id_actual
    POS_X=$(xwininfo -id $id_saved | grep "Absolute upper-left X:" | sed 's/.*:  //')
    POS_Y=$(xwininfo -id $id_saved | grep "Absolute upper-left Y:" | sed 's/.*:  //')
    WINDOW_HEIGHT=$(xwininfo -id $id_saved | grep Height | sed 's/.*: //')
    DESKTOP_HEIGHT=$(xdpyinfo | grep dimensions | sed -r 's/^[^0-9]*([0-9]+x[0-9]+).*$/\1/' | cut -d "x" -f2)
    MIN_Y=26
    MAX_Y=$((DESKTOP_HEIGHT-WINDOW_HEIGHT))
    # MAX_Y=$(xdpyinfo | grep dimensions | sed -r 's/^[^0-9]*([0-9]+x[0-9]+).*$/\1/' | cut -d "x" -f2)
   
    id_actual=$(xdotool getmouselocation | awk -F'window:' '{print $2}')

    if [ "$id_actual" == "$id_saved" ]; then
        echo "I'm jumping" | osd_cat -A center -p middle -f -adobe-helvetica-bold-r-*-*-$font_width-120-*-*-*-*-*-* -cgreen -d1 -O3 &
        if (( $POS_Y < 30 )); then
            xdotool windowmove $id_saved $POS_X $MAX_Y
        else
            xdotool windowmove $id_saved $POS_X $MIN_Y
        fi
    fi

    sleep 0.3
done

fi

```
